src/views/mypage/settings/ui/MySettings.tsx

```diff
@@ -1,12 +1,32 @@
+'use client';
+
+import { axiosDeleteUser } from '@/shared/api';
 import { DeleteAccountButton, ReferralCodeSection } from '@/widgets/mypage';
+import { redirect } from 'next/navigation';
 import React from 'react';
 
 const MySettings = () => {
+  const handleDeleteAccount = async () => {
+    const proceed = confirm(
+      '정말로 회원 탈퇴를 하시겠습니까? 회원 탈퇴 후에는 복구할 수 없습니다.\n\n탈퇴를 원하시면 확인을 눌러주세요.'
+    );
+
+    if (proceed) {
+      const result = await axiosDeleteUser<void>().then(() => true);
+
+      if (result) {
+        redirect('/home');
```

Comment:
```suggestion
       router.push('/home')
```
redirect()는 서버 컴포넌트에서만 사용할 수 있습니다.